Prox/Prox/Utilities/LocationMonitor.swift

```diff
@@ -141,7 +141,10 @@ class LocationMonitor: NSObject {
 
 extension LocationMonitor: LocationProvider {
     func getCurrentLocation() -> CLLocation? {
-        return currentLocation ?? locationManager.location
+        if currentLocation == nil {
+            currentLocation = self.locationManager.location
```

Comment:
iirc, this location may not be have been processed by the locationManager delegate yet – are we okay with that? Are there any assumptions in the code that could cause problems?